src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts

```diff
@@ -59,11 +60,16 @@ export class AdminSearchPageComponent {
         this.statusMessageService.showWarningToast('No results found.');
         this.instructors = [];
         this.students = [];
-      } else {
-        this.instructors = resp.instructors;
-        this.students = resp.students;
-        this.hideAllInstructorsLinks();
-        this.hideAllStudentsLinks();
+        return;
+      }
+
+      this.instructors = resp.instructors;
+      this.students = resp.students;
+      this.hideAllInstructorsLinks();
+      this.hideAllStudentsLinks();
+      if (this.instructors.length + this.students.length >= this.maxResultsToShow) {
```

Comment:
This is not how it works. There can be 50 instructors and 50 students returned each, not 50 max for the total of both.

Reply:
To clarify, would this mean that a total of 100 search results should be shown? If not, is there a priority to follow? (for e.g. exhaust list of instructors first before students etc)

Reply:
Yes, notice there are 2 queries from this:

```java
  searchStudents(searchKey: string, entityType: string): Observable<Students> {
    const paramMap: { [key: string]: string } = {
      searchkey: searchKey,
      entitytype: entityType,
    };
    return this.httpRequestService.get(ResourceEndpoints.SEARCH_STUDENTS, paramMap);
  }

  searchInstructors(searchKey: string): Observable<Instructors> {
    const paramMap: { [key: string]: string } = {
      searchkey: searchKey,
    };
    return this.httpRequestService.get(ResourceEndpoints.SEARCH_INSTRUCTORS, paramMap);
  }
  ```
  
 The backend has a limit for both, so 2x where x is the backend limit (50 in this case)